File: app/src/test/java/com/coinninja/coinkeeper/model/dto/AddressDTOTest.java
package com.coinninja.coinkeeper.model.dto;

import com.coinninja.coinkeeper.model.db.Address;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

import static org.junit.Assert.assertEquals;

@RunWith(MockitoJUnitRunner.class)
public class AddressDTOTest {

    AddressDTO addressDTO;
    String pubKey = "akjsdha3kjrhaecnb3rkjbef";

    @Before
    public void setUp() {
        Address address = new Address();
        address.setIndex(44);
        address.setChangeIndex(0);
        addressDTO = new AddressDTO(address, pubKey);
    }

    @After
    public void tearDown() {
        addressDTO = null;
    }

    @Test
    public void test_display_derivation_path() {
        assertEquals(addressDTO.getDerivationPath(), "M/49/0/0/0/44");
    }
}

File: app/src/test/java/com/coinninja/coinkeeper/model/dto/AddressDTOTest.kt
package com.coinninja.coinkeeper.model.dto

import app.coinninja.cn.libbitcoin.model.DerivationPath
import com.coinninja.coinkeeper.model.db.Address
import com.google.common.truth.Truth.assertThat
import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever

import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.junit.MockitoJUnitRunner


@RunWith(MockitoJUnitRunner::class)
class AddressDTOTest {

    @Test
    fun test_display_derivation_path() {
        val pubKey = "akjsdha3kjrhaecnb3rkjbef"
        val addr = "-- address --"
        val address: Address = mock()
        whenever(address.derivationPath).thenReturn(DerivationPath(49, 0, 0, 0, 44))
        whenever(address.address).thenReturn(addr)
        val addressDTO = AddressDTO(address, pubKey)
        assertThat(addressDTO.derivationPath).isEqualTo("M/49/0/0/0/44")
        assertThat(addressDTO.address).isEqualTo(addr)
        assertThat(addressDTO.uncompressedPublicKey).isEqualTo(pubKey)
    }
}